modules/docs/docgen/spec/docParser.spec.ts

```diff
@@ -0,0 +1,1419 @@
+      expect(docs).toHaveProperty('0.type.right.value', 'Node');
+    });
+
+    it('should handle types that include other symbols when that symbol is exported', () => {
+      const program = createProgramFromSource(`
+        export type Foo = "foo";
+        export type Bar = Foo;
+      `);
+      const docs = parse(program, 'test.ts');
+      expect(docs).toHaveProperty('1.name', 'Bar');
+      expect(docs).toHaveProperty('1.type.kind', 'type');
+      expect(docs).toHaveProperty('1.type.value.kind', 'symbol');
+      expect(docs).toHaveProperty('1.type.value.name', 'Foo');
+    });
+
+    it('should handle types that include other symbols when that symbol is NOT exported', () => {
+      const program = createProgramFromSource(`
+        type Foo<T> = T;
+        export type Bar = Foo<'foo'>;
+      `);
+      const docs = parse(program, 'test.ts');
+      expect(docs).toHaveProperty('0.name', 'Bar');
+      expect(docs).toHaveProperty('0.type.kind', 'type');
+      expect(docs).toHaveProperty('0.type.value.kind', 'string');
+      expect(docs).toHaveProperty('0.type.value.value', 'foo');
+    });
+
+    it('should handle functions with required parameters', () => {
+      const program = createProgramFromSource(`
+        export function myFoo(input: string): boolean {
+          return false
+        }
+      `);
+      const docs = parse(program, 'test.ts');
+      expect(docs).toHaveProperty('0.name', 'myFoo');
+      expect(docs).toHaveProperty('0.type.kind', 'function');
+      expect(docs).toHaveProperty('0.type.parameters.0.kind', 'parameter');
+      expect(docs).toHaveProperty('0.type.parameters.0.name', 'input');
+      expect(docs).toHaveProperty('0.type.parameters.0.defaultValue', undefined);
+      expect(docs).toHaveProperty('0.type.parameters.0.required', true);
+      expect(docs).toHaveProperty('0.type.parameters.0.type', {
+        kind: 'primitive',
+        value: 'string',
+      });
+    });
+
+    it('should handle functions with optional parameters', () => {
+      const program = createProgramFromSource(`
+        /** jsdoc for myFoo
+         * @deprecated
+         * @example
+         * // my example
+         */
+        export function myFoo(input?: string): boolean {
+          return false
+        }
+      `);
+      const docs = parse(program, 'test.ts');
+      expect(docs).toHaveProperty('0.name', 'myFoo');
+      expect(docs).toHaveProperty('0.type.kind', 'function');
+      expect(docs).toHaveProperty('0.type.parameters.0.kind', 'parameter');
+      expect(docs).toHaveProperty('0.type.parameters.0.name', 'input');
+      expect(docs).toHaveProperty('0.type.parameters.0.defaultValue', undefined);
+      expect(docs).toHaveProperty('0.type.parameters.0.required', false);
+      expect(docs).toHaveProperty('0.type.parameters.0.type', {
+        kind: 'primitive',
+        value: 'string',
+      });
+    });
+
+    it('should handle functions with optional parameters', () => {
```

Comment:
This has the same description as the previous spec. Maybe:
```suggestion
    it('should handle functions with optional parameters that have a default value', () => {
```

Reply:
Nice